text/0000-app-image-extensions.md

```diff
@@ -21,44 +23,42 @@ Allowing buildpacks to install OS packages dynamically during build would drasti
 Mixins already allow buildpack authors to create buildpacks that depend on an extended set of OS packages without affecting build time.
 However, it is not uncommon for application code to depend on OS packages.
 
-Given that app authors and platform maintainers experience increased build time directly, extending the stack at the app author's or platform maintainer's request may add flexibility without sacrificing performance.
+Given that app authors and platform maintainers experience increased build time directly, extending the stack at the app author's or platform maintainer's request may add flexibility without sacrificing performance. At the same time, we want the CNB interface to be highly orthogonal, which is why we've strived to "make everything a buildpack". Allowing some buildpacks to run with privileges would give users the flexibility they expect based on their experience with other tools like `Dockerfile`.
+
+The advantages of using a buildpack for privileged operations are the same as for unprivileged operations: they are composable, fast (caching), modular, reuseable, and safe.
+
+As an example, consider the use case of Acme.com. They have three teams that want to contibute privileged layers to every image built at the company. Each of these teams wants to manage their own artifacts and the mechanisms for installing them, but none of these teams are in control of the base image. The teams have special logic for when their layers should be added to an image; in some cases it's the precense of a Node.js app, and in others it's the use of Tomcat, etc. In the past, these teams were trying to contribute their layers by selectively adding them to every `Dockerfile` for every base image in the company, but this doesn't scale well. With Root Buildpacks, each team can manage their own artifacts, release cadence, and add logic to `bin/detect` to ensure the layers are added when needed.
 
 # What it is
 [what-it-is]: #what-it-is
 
+Stack image creators and buildpack users can use buildpacks to extend their build and/or run images.
+
 ## Specification
 
-Stack image creators may add the following executables to their run and/or build images:
+Introduce a boolean `privileged` key in the `[buildpack]` table of `buildpack.toml`, which is defined as follows:
 
-Extension:
 ```
-/cnb/image/build/extend (build-metadata-toml-file) (cache-dir) | cwd: /
-/cnb/image/run/extend (run-metadata-toml-file) (cache-dir) | cwd: /
+[buildpack]
+privileged = <boolean (default=false)>
 ```
-Status:
+
+When `privileged` is set to `true`, the lifecycle will run this buildpack as the `root` user.
+
+For each Root Buildpack, the lifecycle will mount an overlay filesystem before the buildpack's build phase executes (excluding `/tmp`, `/cnb`, and `/layers`). The buildpack can then create layers normally by writing to the `<layers>/` directory the same way an unprivileged buildpack would. However, a new `paths` key would be added to the `<layers>/<layer>.toml` schema:
```

Comment:
This means that some layers would be owned by CNB_UID and some layers would be owned by root. This will create complexity and complicate the security posture (especially if, e.g., SUID is allowed). Can we avoid these special buildpacks having two responsibilities?

Reply:
What are the "two responsibilities"?